src/main/java/hudson/plugins/textfinder/TextFinderPublisher.java

```diff
@@ -128,24 +128,29 @@ private void findText(Run<?, ?> run, FilePath workspace, TaskListener listener)
             boolean foundText = false;
 
             if (alsoCheckConsoleOutput) {
-                logger.println("Checking console output");
+                // Do not mention the pattern we are looking for to avoid false positives
+                logger.println("Looking for a specific pattern in the console output");
```

Comment:
Looks great! Thanks for doing this. There's only one minor comment I have. The wording of the message "Looking for a specific pattern in the console output" seems potentially confusing to the end user. To me, this message raises the question: "What _is_ that specific pattern?" Of course, we can't tell the user what the specific pattern is, because that would lead to a false positive, as your comment states. How about if we changed the logic such that instead of printing "Looking for a specific pattern in the console output", we printed "Searched for pattern X in the console output" _after_ performing the search? That way, we'd avoid the false positive. We'd probably want to change the behavior for files to do the same thing, for consistency. Then we'd be able to print the pattern in the message in both cases. Let me know what you think of this idea. If you don't think it's worth the additional work, then I'm fine with this change as-is.

Reply:
Thanks for taking the time on thinking about this!
I also went through this a couple of times, I really wanted to show the pattern but felt it required some user feedback BEFORE starting looking.
I was thinking on saving the console log, print message and then do the search on the saved content but the size might be too big in memory in certain situations.
I will try to apply your suggestion, adding a message BEFORE the search so the user knows what is happening.